src/parser/contexts.h

```diff
@@ -488,18 +495,26 @@ struct ParseDeclsCtx : NullTypeParserCtx, NullInstrParserCtx {
   std::vector<DefPos> typeDefs;
   std::vector<DefPos> subtypeDefs;
   std::vector<DefPos> funcDefs;
+  std::vector<DefPos> tableDefs;
   std::vector<DefPos> memoryDefs;
   std::vector<DefPos> globalDefs;
+  std::vector<DefPos> elemDefs;
   std::vector<DefPos> dataDefs;
   std::vector<DefPos> tagDefs;
 
   // Positions of typeuses that might implicitly define new types.
   std::vector<Index> implicitTypeDefs;
 
+  // Map table indices to the indices of their implicit, in-line element
+  // segments.
+  std::unordered_map<Index, Index> implicitElemIndices;
```

Comment:
Because implicit data segments do not contain types or instructions, so we never need to find them again after the initial parser phase where they are first created. In contrast, we need to find implicit elem segments again during `ParseModuleTypes` to set the type of the element segment based on the type of its table, then we need to find it again during `ParseDefs` to parse the instructions in the segment.

Reply:
I see, thanks - maybe add a comment here for how this is used, that could state that?